minio/known_error.py

```diff
@@ -0,0 +1,369 @@
+              'request does not match an existing version.'
+
+class NotImplemented(KnownResponseError):
+    message = 'A header you provided implies functionality ' \
+              'that is not implemented.'
+
+class NotSignedUp(KnownResponseError):
+    message = 'Your account is not signed up.'
+
+class NoSuchBucketPolicy(KnownResponseError):
+    message = 'The specified bucket does not have a bucket policy.'
+
+class OperationAborted(KnownResponseError):
+    message = 'A conflicting conditional operation is currently in ' \
+              'progress against this resource. Try again.'
+
+class PermanentRedirect(KnownResponseError):
+    message = 'The bucket you are attempting to access must be addressed ' \
+              'using the specified endpoint. Send all future requests ' \
+              'to this endpoint.'
+
+class PreconditionFailed(KnownResponseError):
+    message = 'At least one of the preconditions you specified did not hold.'
+
+class Redirect(KnownResponseError):
+    message = 'Temporary redirect.'
+
+class RestoreAlreadyInProgress(KnownResponseError):
+    message = 'Object restore is already in progress.'
+
+class RequestIsNotMultiPartContent(KnownResponseError):
+    message = 'Bucket POST must be of the enclosure-type multipart/form-data.'
+
+class RequestTimeout(KnownResponseError):
+    message = 'Your socket connection to the server was not read ' \
+              'from or written to within the timeout period.'
+
+class RequestTimeTooSkewed(KnownResponseError):
+    message = 'The difference between the request time and the ' \
+              'server\'s time is too large.'
+
+class RequestTorrentOfBucketError(KnownResponseError):
+    message = 'Requesting the torrent file of a bucket is not permitted.'
+
+class SignatureDoesNotMatch(KnownResponseError):
+    message = 'The request signature we calculated does not match the ' \
+              'signature you provided.'
+
+class ServiceUnavailable(KnownResponseError):
+    message = 'Reduce your request rate.'
+
+class SlowDown(KnownResponseError):
+    message = 'Reduce your request rate.'
+
+class TemporaryRedirect(KnownResponseError):
+    message = 'You are being redirected to the bucket while DNS updates.'
+
+class TokenRefreshRequired(KnownResponseError):
+    message = 'The provided token must be refreshed.'
+
+class TooManyBuckets(KnownResponseError):
+    message = 'You have attempted to create more buckets than allowed.'
+
+class UnexpectedContent(KnownResponseError):
+    message = 'This request does not support content.'
+
+class UnresolvableGrantByEmailAddress(KnownResponseError):
+    message = 'The email address you provided does not match any account ' \
+              'on record.'
+
+class UserKeyMustBeSpecified(KnownResponseError):
+    message = 'The bucket POST must contain the specified field name. ' \
+              'If it is specified, check the order of the fields.'
+
+known_errors = {
+    'AccessDenied': AccessDenied,
```

Comment:
You could build the dictionary by iterating a copy of the dict `locals()` (filtering only for the instances of KnownResponseError): it would avoid you to copy all the exceptions twice, it would avoid typos and it would be automatically kept up-to-date